crates/bevy_utils/src/lib.rs

```diff
@@ -233,3 +234,26 @@ impl<K: Hash + Eq + PartialEq + Clone, V> PreHashMapExt<K, V> for PreHashMap<K,
         }
     }
 }
+
+/// A type which calls a function when dropped.
+/// This can be used to ensure that cleanup code is run even in case of a panic.
+pub struct OnDrop<F: FnOnce()> {
+    callback: ManuallyDrop<F>,
+}
```

Comment:
The previous (private) version of this type used `F: FnMut()`, which made working with owned values complicated. Since we need unsafe code in order to make this work using `FnOnce()`, I figured it's worth going the extra mile and making this public.

Reply:
Very clever: I like this. It would be nice to see a small doc test here.